### DISCUSSION TITLE
How to switch on debug logging?

### DISCUSSION THREAD

**Discussion by ronaldploeger** [artilleryio/artillery#958] [Artillery]
How does one switch on debug logging?

Best,
Ronald

**Comment by hassy** [comment]
hi @ronaldploeger!👋 if you're running HTTP tests you can set the `DEBUG` environment variable to `http,http:request,http:response` to see debug information, e.g.:

```sh
DEBUG=http,http:request,http:response artillery run my-script.yml
```

**Reply by chiptus** [reply]
what can I do to run for socket io?

**Reply by hassy** [reply]
`DEBUG=socketio` will enable some debug logging for Socket.io @chiptus

**Comment by roshannair89** [comment]
Hi @hassy I am using VS Code terminal for my tests and I am trying to run the _DEBUG=http,http:request,http:response artillery run my-script.yml_ command as shown in the below image, but no luck! What am I missing?

![image](https://user-images.githubusercontent.com/89486127/153316048-7842fae2-13a8-425b-8616-16dc080c4c21.png)

**Reply by hassy** [reply]
it looks like you're on Windows @roshannair89 - setting an environment variable there requires a slightly different syntax, please see the example here: https://www.artillery.io/docs/guides/guides/http-reference#debugging

**Reply by roshannair89** [reply]
Thanks Hassy! Related to this, in case I am load testing a scenario with multiple HTTP request s chained one after the other is there a way to generate a more detailed HTML report that shows how each individual services did? For example, if I had a GET followed by POST and if the GET timed out for a few of my virtual users then the current HTML Report only shows me x users failed (ETIMEDOUT) but there is no way for me to see which service actually failed

**Reply by efabisze** [reply]
I tried the debugging by sending "set DEBUG=https*" followed by running the test but i do not see any logging from the requests or any responses, am i missing something here? @roshannair89 @hassy

**Reply by hassy** [reply]
@efabisze - the value should be `http*` rather than `https*`

**Reply by hassy** [reply]
@roshannair89 sorry missed your question. This plugin should help you analyze endpoint/service specific failures: https://www.artillery.io/docs/guides/plugins/plugin-metrics-by-endpoint

**Reply by efabisze** [reply]
@hassy 
Thanks this was my mistake

**Reply by newuserbk** [reply]
@ronaldploeger @hassy @efabisze @roshannair89 
I am using VS Code terminal for my artillery yml tests and I am trying to run the DEBUG on windows machine
I followed below step-
set  DEBUG=http* 
DEBUG=http:request artillery run test.yml
DEBUG=http:response artillery run test.yml
DEBUG=http:request,http:response artillery run test.yml

But I am getting below error-
<img width="709" alt="image" src="https://github.com/artilleryio/artillery/assets/39767148/a1365fcf-0f09-4ddd-8bbe-947ebb531dd9">


I also tried to set values in package.json but still getting same error. Am I missing something ?
`{
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1",
    "start": "set DEBUG=artillery_load_test & node app.js"
  },
  "dependencies": {
    "artillery-plugin-expect": "^2.3.2",
    "artillery-plugin-metrics-by-endpoint": "^1.2.0"
  },
  "devDependencies": {
    "artillery": "^2.0.0-37",
    "faker": "^5.5.3"
  }
}
`

**Reply by bernardobridge** [reply]
Hi @newuserbk 👋, you've tried using the format for Linux and Command Prompt in Windows, but it seems you're using a Powershell prompt. Our docs section explain how to turn it on in a Powershell prompt (`$Env:DEBUG = 'http'`): https://www.artillery.io/docs/reference/engines/http#debugging

**Reply by newuserbk** [reply]
@bernardobridge 
I tried on PowerShell on windows and still I am not seeing request and response in console
<img width="623" alt="image" src="https://github.com/artilleryio/artillery/assets/39767148/2ac5e5a0-ca81-42a2-b4a0-7beda7a5414b">

I also tried on windows platform on command prompt and still not seeing logs
<img width="704" alt="image" src="https://github.com/artilleryio/artillery/assets/39767148/439b315a-56ae-45e0-b67e-c849c6c3fc3c">

**Reply by bernardobridge** [reply]
Hi @newuserbk . For displaying request and response, you need to enable the values `http:request` and `http:response` for the DEBUG variable. You should be able to set them as comma-separated values (e.g. `http,http:request,http:response`).

**Reply by newuserbk** [reply]
@bernardobridge 
I tried by setting set DEBUG:http,http:request,http:respone but still not able to see logs
C:\project>set DEBUG:http,http:request,http:respone
C:\project>artillery run graphql.yml

Am I missing something or any mistake am I doing?
<img width="707" alt="image" src="https://github.com/artilleryio/artillery/assets/39767148/8114765b-3930-4f37-b6d4-b70d2ecc33f7">

Can you please help with syntax on window command and windows powershell, thanks

**Reply by bernardobridge** [reply]
Hi @newuserbk ! Sorry it took a bit to get back to you - it turns out we had a bug in the logging , and were not printing the errors coming from directly `got` (our http library).

I've now fixed that and it's available in our canary version (`artillery@2.0.0-38-a4feb0d`). So if you install that version instead of the version you're currently on, and set the debug logs in the same way explained above, you should be able to see an error coming from the http library that will give you more information about what might be wrong in the test script.
